broker/broker-undelivered-messages-ui/src/main/webapp/reducer.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import {combineReducers} from "redux"
+
+const messages = (messages = [], {type, checked = false, expanded = false, id, message, ids}) => {
+    switch (type) {
+        case 'ADD_MESSAGE':
+            return [...messages, {checked, ...message}]
```

Comment:
I am not sure how large the store is getting at runtime, but if it's getting large enough, we should consider using an immutable [List](https://facebook.github.io/immutable-js/docs/#/List) to improve performance.